Neos.Flow/Classes/Persistence/Doctrine/ObjectValidationListener.php

```diff
@@ -0,0 +1,127 @@
+<?php
+namespace Neos\Flow\Persistence\Doctrine;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Doctrine\ORM\UnitOfWork;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Persistence\Exception\ObjectValidationFailedException;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\Reflection\ClassSchema;
+use Neos\Utility\ObjectAccess;
+use Neos\Flow\Reflection\ReflectionService;
+use Neos\Utility\TypeHandling;
+use Neos\Flow\Validation\ValidatorResolver;
+
+/**
+ * Flow's Doctrine PersistenceManager
+ *
+ * @Flow\Scope("singleton")
+ * @api
+ */
+class ObjectValidationListener
+{
+    /**
+     * @Flow\Inject
+     * @var ObjectManager
+     */
+    protected $entityManager;
+
+    /**
+     * @Flow\Inject
+     * @var ValidatorResolver
+     */
+    protected $validatorResolver;
+
+    /**
+     * @Flow\Inject
+     * @var ReflectionService
+     */
+    protected $reflectionService;
+
+    /**
+     * @Flow\Inject
+     * @var PersistenceManagerInterface
```

Comment:
Might as well annotate the doctrine persistence manager implementation as it won't work with any other persistence manager anyway.